lib/json_schema/version_event_file_name.rb

```diff
@@ -1,24 +1,23 @@
 # frozen_string_literal: true
 
-require "initialize_with_config"
-
 module JsonSchema
   class VersionEventFileName
-    include InitializeWithConfig
-    required_config :version, :event
+    class << self
+      def call(schema_root: "etc/schema", schema_path: "ecf/participant_declarations", schema_file: "request_schema.json", version: "1.0", event: :create)
+        new(schema_path: schema_path, schema_root: schema_root, schema_file: schema_file).call(version: version, event: event)
+      end
+    end
 
-    def call
-      Rails.root.join(schema_root, version.to_s, schema_path, event.to_s, schema_file)
+    def call(version:, event:)
+      Rails.root.join(@schema_root, version.to_s, @schema_path, event.to_s, @schema_file)
     end
 
   private
 
-    def default_config
-      {
-        schema_root: "etc/schema",
-        schema_path: "ecf/participant_declarations",
-        schema_file: "request_schema.json",
-      }
+    def initialize(schema_root: "etc/schema", schema_path: "ecf/participant_declarations", schema_file: "request_schema.json")
+      @schema_root = schema_root
+      @schema_path = schema_path
+      @schema_file = schema_file
     end
```

Comment:
I quite like having a list of all parameters that you can pass when creating an object in `initialize` which lives on the class. It helps ensure you are not passing something that's unused, and it makes it easier to spot if the class is missing some info.

If we want to have a re-usable way of handling 'missing parameter' errors, we could do it with a helper method, say `validate_parameters_exist`, called at the end of `initialize` with the parameters we want to enforce. I guess it would need to be aware of whether some `@` variables are defined, but I hope Ruby lets you check that.

Reply:
It's a bit counter to the "Pass no more than four parameters into a method. Hash options are parameters." from the DfE Technical guidelines link. Personally, I prefer to pass an object which responds (and can be delegated to), which adheres more to the Liskov Substitution Principle.